src/rebuild/scan.c

```diff
@@ -536,6 +526,14 @@ rebuild_scan_leader(void *data)
 	struct rebuild_pool_tls	  *tls;
 	int			   rc;
 
+	D_DEBUG(DB_REBUILD, DF_UUID "check resync %u < %u\n",
+		DP_UUID(rpt->rt_pool_uuid), rpt->rt_pool->sp_dtx_resync_version,
+		rpt->rt_rebuild_ver);
+
+	/* Wait for dtx resync to finish */
+	while (rpt->rt_pool->sp_dtx_resync_version < rpt->rt_rebuild_ver)
+		ABT_thread_yield();
```

Comment:
The only disadvantage for yield is this ULT will be kept scheduling before it is ready, but 
oh, this is very rare case.
So I would prefer to keep the code simple here, instead of adding condition and wakeup.

Reply:
I am not sure it is right. Consider the case that the first dtx resync is in-handling, assume that the rpt->rt_pool->sp_dtx_resync_version is 5. The first rebuild rpt->rt_rebuild_ver is also 5. We expect that the rebuild will wait until the first dtx resync done. But at that time, pool map is refreshed again, the second dtx resync is triggered, then the rpt->rt_pool->sp_dtx_resync_version is increased to 6, that will wakeup the first waiting rebuild logic. But the first dtx resync (with ver 5) is not done yet. That is unexpectedly.

Reply:
rpt->rt_pool->sp_dtx_resync_version will not be upgraded to 6, until dtx resync is finished.  So if dtx resync 6 is finished, then I assume all previous version (include 5) are finished anyway. so it is safe to go ahead for rebuild?

Reply:
How can you guarantee that the pool map version will not be upgraded to 6 during dtx resync for PM 5?